### PR TITLE
Update plex.markdown

### PR DESCRIPTION
**Pull request by TheDK**
Added information about DNS Rebind and usage on internet connections with dynamic public IP addresses.

## Proposed change
Update of Plex documenation to account for DNS Rebind protection and usage on internet connections with dynamic IP address.

## Type of change
- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist
- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @TheDK, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant.io#16355).

Once you do that we will be able to review and accept this pull request.

Thanks!

File: source/_integrations/plex.markdown
Comment (frenck) [inline source/_integrations/plex.markdown]:
```suggestion
During setup, the integration will check all possible ways to connect to your Plex server(s) - i.e., the local network IP address or the public IP address of your internet connection, HTTP or HTTPS, directly or via the domain `plex.direct`, and chose local over public and secure over insecure. You can check your HA log to see which connection was chosen. If your internet router uses DNS rebind protection, the direct connection to the local address via `plex.direct` will fail (see [Plex documentation](https://support.plex.tv/articles/206225077-how-to-use-secure-server-connections/) - scroll way down to DNS Rebinding). To avoid this, configure your router to allow the domain `plex.direct`, if possible.
```

File: source/_integrations/plex.markdown
Comment (frenck) [inline source/_integrations/plex.markdown]:
```suggestion
Please note that if the connection using the public IP address was chosen and you have a dynamic public IP or your public IP changes connecting to the server will fail. In that case, use the manual configuration (see below) to ensure that a local IP is being used for the connection.
```

**Review by frenck — DISMISSED** [review]
Some small suggestions. I'll ping the maintainer for a check.

**Comment by TheDK** [comment]
Thanks for the grammar check - apparently I have to work on my punctuation...  Do I have to do anything to include the suggestions?

**Comment by jjlawren** [comment]
Thanks @TheDK. After reading through this I have an idea for a different approach in the config flow that may make (most of) this doc change unnecessary. Can we put this on hold while I see if the other method is workable?

**Comment by TheDK** [comment]
Sure - looking forward to what you have in mind!

**Comment by jjlawren** [comment]
@TheDK now that the behavior has changed (as discussed in https://github.com/home-assistant/core/issues/45419), do you still think this is necessary?

**Comment by TheDK** [comment]
Why not add the info about how it works for transparency? I could update the text reflecting the changes...

**Comment by jjlawren** [comment]
Sorry, I meant to follow up on this months ago and totally forgot. Let me make a few suggestions

File: source/_integrations/plex.markdown
Comment (jjlawren) [inline source/_integrations/plex.markdown]: The change in https://github.com/home-assistant/core/pull/45878 should make this note obsolete. I'm also hesitant to encourage use of the manual configuration.
```suggestion
```

File: source/_integrations/plex.markdown
Comment (jjlawren) [inline source/_integrations/plex.markdown]:
```suggestion
During setup, the integration will check all possible ways to connect to your Plex server(s) - i.e., local or public addresses, HTTP or HTTPS, by IP or using a subdomain of `plex.direct`, or by using a Plex relay if all other methods fail. The integration will prefer local over public and secure over insecure, in that order. The selected address is shown on the Plex card on the Integrations page.

If your router enforces DNS rebind protection, connections to the local `plex.direct` hostname may fail (see [Plex documentation](https://support.plex.tv/articles/206225077-how-to-use-secure-server-connections/#dnsrebinding)). To avoid this, configure your router to allow DNS rebinding for `plex.direct` by following the instructions in the documentation link.
```

**Review by jjlawren — CHANGES_REQUESTED** [review]

File: source/_integrations/plex.markdown
Comment (jjlawren) [reply source/_integrations/plex.markdown]: I've suggested to remove this paragraph instead.

**Comment by TheDK** [comment]
Hey guys, thanks for picking it up again! I just did it full run on my test environment - the local IP was correctly chosen (strangely with http, not https (http://192.168.23.11:32400) - even though we state above the secure is chosen over inscure connections?), but imo this is not critical for this. --> Very good solution, thanks for the efforts! Please close if you don't have anymore comments...

**Comment by jjlawren** [comment]
> I just did it full run on my test environment - the local IP was correctly chosen (strangely with http, not https even though we state above the secure is chosen over inscure connections?

Did you see anything in the logs about the local secure connection failing? Maybe DNS lookup still failed. Either way, this is still a successful result.

**Review by jjlawren — APPROVED** [review]
This seems like good context for users running into issues 👍

**Comment by jjlawren** [comment]
@frenck anything else here? I think it looks good to go.

**Pull request merged**
